crates/types/src/config/mod.rs

```diff
@@ -244,6 +245,26 @@ impl Configuration {
     pub fn dump(&self) -> Result<String, GenericError> {
         Ok(toml::to_string_pretty(self)?)
     }
+
+    /// Checks whether the given configuration is valid. Returns an [`InvalidConfigurationError`]
+    /// it if is not valid.
+    pub fn validate(&self) -> Result<(), InvalidConfigurationError> {
+        if self
+            .common
+            .force_node_id
+            .is_some_and(|force_node_id| force_node_id == PlainNodeId::new(0))
+        {
+            return Err(InvalidConfigurationError::ForceNodeIdZero);
+        }
+
+        Ok(())
+    }
```

Comment:
An alternative is to use NonZero<u32> for this field, but you wouldn't be able to control the error message (unless you do that through the shadow configuration.

Reply:
True. I wanted to give a more explanatory error message to users that's why I went with the `validate` approach. I don't have a strong opinion here and can move the logic to the `CommonOptionsShadow`. One advantage of using the `validate` method could be that we could use the `CodedErrors` and log it accordingly.